tests/MongoDB.Bson.Tests/IO/JsonReaderTests.cs

```diff
@@ -45,12 +45,14 @@ public class JsonReaderTests
             {
                 var result = new List<BsonDocument>();
 
+                jsonReader.State.Should().Be(BsonReaderState.Initial);
                 while (!jsonReader.IsAtEndOfFile())
                 {
                     jsonReader.ReadStartDocument();
                     var name = jsonReader.ReadName();
                     var value = jsonReader.ReadInt32();
                     jsonReader.ReadEndDocument();
+                    jsonReader.State.Should().Be(BsonReaderState.Initial);
```

Comment:
Throughout this test file I've made sure that we consistently check both State and IsAtEndOfFile.